test/framework/src/main/java/org/elasticsearch/test/rest/client/RestClient.java

```diff
@@ -284,7 +288,42 @@ protected CloseableHttpClient createHttpClient(Settings settings) throws IOExcep
                 SSLContext sslcontext = SSLContexts.custom()
                         .loadTrustMaterial(keyStore, null)
                         .build();
-                sslsf = new SSLConnectionSocketFactory(sslcontext);
+                sslsf = new SSLConnectionSocketFactory(sslcontext, new X509HostnameVerifier() {
+
+                    // We need to wrap the default verifier for HttpClient since we use an older version and the following issue is not
+                    // fixed in a released version yet https://issues.apache.org/jira/browse/HTTPCLIENT-1698
+                    // TL;DR we need to strip '[' and ']' from IPv6 addresses if they come from a URL
+                    private final X509HostnameVerifier verifier = new BrowserCompatHostnameVerifier();
```

Comment:
`BrowserCompatHostnameVerifier` is deprecated in later Apache HTTP client releases.

Additionally, it does the **wrong thing** for wildcard certs. I think we should use the non-deprecated strict hostname verifier